vyper/functions/signature.py

```diff
@@ -69,17 +72,30 @@ def process_arg(index, arg, expected_arg_typelist, function_name, context):
                 return sub
         else:
             # Does not work for unit-endowed types inside compound types, e.g. timestamp[2]
-            parsed_expected_type = context.parse_type(ast.parse(expected_arg).body[0].value, 'memory')
+            parsed_expected_type = context.parse_type(
+                ast.parse(expected_arg).body[0].value,
+                'memory',
+            )
             if isinstance(parsed_expected_type, BaseType):
                 vsub = vsub or Expr.parse_value_expr(arg, context)
+
+                is_valid_integer = (
+                    (
+                        expected_arg in ('int128', 'uint256') and isinstance(vsub.typ, BaseType)
+                    ) and (
+                        vsub.typ.typ in ('int128', 'uint256') and vsub.typ.is_literal
+                    ) and (
+                        SizeLimits.in_bounds(expected_arg, vsub.value)
+                    )
+                )
+
                 if is_base_type(vsub.typ, expected_arg):
                     return vsub
-                elif expected_arg in ('int128', 'uint256') and \
-                     isinstance(vsub.typ, BaseType) and \
-                     vsub.typ.typ in ('int128', 'uint256') and \
-                     vsub.typ.is_literal and \
-                     SizeLimits.in_bounds(expected_arg, vsub.value):
+                elif is_valid_integer:
                     return vsub
+                # TODO: missing final else clasue allowing implicit exit of
+                # this block without hitting any of the logic.  Is this
+                # intended?
```

Comment:
@jacqueswww, should evaluate this

Reply:
@davesque remove this TODO, if anything fails the exception below will pick it up.

Reply:
@davesque I believe `return arg.id` on line 109 is unreachable code as well.